api/resolvers/wallet.js

```diff
@@ -73,9 +73,9 @@ export default {
     connectAddress: async (parent, args, { lnd }) => {
       return process.env.LND_CONNECT_ADDRESS
     },
-    walletHistory: async (parent, { cursor, inc }, { me, models, lnd }) => {
+    walletHistory: async (parent, { cursor, inc, limit = LIMIT, id }, { me, models, lnd }) => {
       const decodedCursor = decodeCursor(cursor)
-      if (!me) {
+      if (!me && !id) {
```

Comment:
Doesn't this allow anyone to fetch the wallet history of any user?

For example:

```
$ curl -XPOST -H "Content-type: application/json" --data @wallet_history.json https://sn.ekzyis.com/api/graphql
{"data":{"walletHistory":{"facts":[]}}}
```

_wallet\_history.json_:

```json
{
  "query": "query walletHistory($id: Int) { walletHistory(id: $id) { facts { id createdAt factId bolt11 sats satsFee status type description item { id } } } }",
  "variables": {
    "id": 1
  }
}
```

However, I created a new user and funded my account. I then used the id of that new user and expected to see their wallet history but the history in the response was empty :thinking: 

I am too tired at the moment to debug why it's empty; had a long day... will take a look at this PR again after I slept!

Reply:
I'm learning all this as I go, but I believe you might be correct that this exposes the wallet history of all users in a way that it wasn't before. (You need the "inc" variable to get non-empty results.) This is partly why I was questioning whether there's a better way to handle this "me" issue here.

In essence, the problem is that to fulfill its jobs, the worker is querying the GQL API just like anybody else. It needs its own privileged access somehow.

Reply:
> You need the "inc" variable to get non-empty results.

Ah, will check this out when I have more time

> In essence, the problem is that to fulfill its jobs, the worker is querying the GQL API just like anybody else. It needs its own privileged access somehow.

The worker has direct database access. Whatever you want to do, you don't need to go through the GraphQL API.

Reply:
Alright. I've pushed another commit that switches from GraphQL to Postgres queries. This solves the problem with the id parameter (and removes it). So I'll mark this conversation/thread as resolved.